wp-content/themes/goonj-crm/engine/shortcodes.php

```diff
@@ -130,3 +132,86 @@ function goonj_collection_camp_past_data() {
 	get_template_part( 'templates/collection-camp-data' );
 	return ob_get_clean();
 }
+
+function goonj_induction_slot_details() {
+    // Retrieve parameters from the GET request
+    $source_contact_id = intval($_GET['source_contact_id'] ?? 0);
+    $slot_date = $_GET['slot_date'] ?? '';
+    $slot_time = $_GET['slot_time'] ?? '';
+    $inductionType = $_GET['induction_type'] ?? '';
+
+    try {
+        // Fetch the induction activity for the specified source contact
+        $inductionActivity = \Civi\Api4\Activity::get(FALSE)
+            ->addSelect('id', 'activity_date_time', 'status_id:name', 'Induction_Fields.Goonj_Office', 'Induction_Fields.Assign')
+            ->addWhere('source_contact_id', '=', $source_contact_id)
+            ->addWhere('activity_type_id:name', '=', 'Induction')
+            ->execute()
+            ->single();
+		\Civi::log()->info('inductionActivity', ['inductionActivity'=>$inductionActivity]);
+
+        // Exit if no activity found or the status is not "To be Scheduled"
+        if (!$inductionActivity || $inductionActivity['status_id:name'] !== 'To be scheduled') {
+            \Civi::log()->info('No valid activity found or status is not To be Scheduled', ['contact_id' => $source_contact_id]);
+            return;
+        }
+
+		// Combine slot date (d-m-Y) and slot time (h:i A) to form new activity date and time
+		$newActivityDateTime = DateTime::createFromFormat('d-m-Y h:i A', "$slot_date $slot_time");
+		if (!$newActivityDateTime) {
+			\Civi::log()->error('Invalid date/time format', ['slot_date' => $slot_date, 'slot_time' => $slot_time]);
+			return;
+		}
+
+
+        // Update activity date and time and set the status to "Scheduled"
+        \Civi\Api4\Activity::update(FALSE)
+            ->addValue('activity_date_time', $newActivityDateTime->format('Y-m-d H:i:s'))
+            ->addValue('status_id:name', 'Scheduled')
+            ->addValue('Induction_Fields.Mode:name', $inductionType)
+            ->addWhere('id', '=', $inductionActivity['id'])
+            ->execute();
+        \Civi::log()->info('Activity updated successfully', [
```

Comment:
Needed?

Reply:
Yes, for monitoring logs or any bug occur we can check this log